src/doc/src/reference/registries.md

```diff
@@ -320,9 +320,8 @@ visit the registry's website to obtain a token, and Cargo can store the token
 using the [`cargo login`] command, or by passing the token on the
 command-line.
 
-Responses use a 200 response code for both success and errors. Cargo looks at
-the JSON response to determine if there was success or failure. Failure
-responses have a JSON object with the following structure:
+Responses should use the 200 or 404 response code as appropriate. Failure
```

Comment:
I don't think servers need to restrict themselves to 404 for errors. I think they should be free to return the appropriate response (400, 401, 403, 404, etc.) as they see fit. 

Maybe something like this?

```suggestion
Responses use the 200 response code for success.
Errors should use an appropriate response code, such as 404.
Failure
```